improver_tests/ensemble_copula_coupling/test_utilities.py

```diff
@@ -329,177 +310,98 @@ class Test_insert_lower_and_upper_endpoint_to_1d_array(IrisTest):
+        points_data = [self.cube.data[i].flatten() for i in range(3)]
+        self.input_data = np.array(points_data)
 
     def test_percentile_is_dimension_coordinate(self):
         """
-        Test that the result have the expected size for the
-        probabilistic dimension and is generally of the expected size.
-        The array contents is also checked.
+        Test the result is an array with the expected shape and contents.
         """
-        cube = self.current_temperature_forecast_cube
-        plen = len(cube.coord("percentile").points)
-        reshaped_array = restore_non_probabilistic_dimensions(
-            cube.data, cube, "percentile", plen
+        plen = len(self.cube.coord("percentile").points)
+        reshaped_array = restore_non_percentile_dimensions(
+            self.input_data, next(self.cube.slices_over("percentile")), plen
         )
-        self.assertEqual(reshaped_array.shape[0], plen)
-        self.assertEqual(reshaped_array.shape, cube.data.shape)
-        self.assertArrayAlmostEqual(reshaped_array, cube.data)
-
-    def test_if_percentile_is_not_first_dimension_coordinate(self):
-        """
-        Test that the result have the expected size for the
-        probabilistic dimension and is generally of the expected size.
-        The array contents is also checked.
-        """
-        cube = self.current_temperature_forecast_cube
-        cube.transpose([2, 1, 0])
-        plen = len(cube.coord("percentile").points)
-        msg = "coordinate is a dimension coordinate but is not"
-        with self.assertRaisesRegex(ValueError, msg):
-            restore_non_probabilistic_dimensions(cube.data, cube, "percentile", plen)
+        self.assertIsInstance(reshaped_array, np.ndarray)
+        self.assertEqual(reshaped_array.shape, self.cube.data.shape)
+        self.assertArrayAlmostEqual(reshaped_array, self.cube.data)
 
     def test_percentile_is_not_dimension_coordinate(self):
         """
-        Test the array size, if the percentile coordinate is not a dimension
-        coordinate on the cube.
-        The array contents is also checked.
+        Test the array size and contents if the percentile coordinate is not a
+        dimension coordinate on the input cube.
         """
         expected = np.array(
-            [
-                [
-                    [226.15, 237.4, 248.65],
-                    [259.9, 271.15, 282.4],
-                    [293.65, 304.9, 316.15],
-                ]
-            ],
+            [[226.15, 237.4, 248.65], [259.9, 271.15, 282.4], [293.65, 304.9, 316.15]],
             dtype=np.float32,
         )
 
-        cube = self.current_temperature_forecast_cube
-        cube_slice = next(cube.slices_over("percentile"))
-        plen = len(cube_slice.coord("percentile").points)
-        reshaped_array = restore_non_probabilistic_dimensions(
-            cube_slice.data, cube_slice, "percentile", plen
+        cube_slice = next(self.cube.slices_over("percentile"))
```

Comment:
Would it make sense for `cube_slice` to be in `setUp` given that it is effectively used in `test_percentile_is_dimension_coordinate` and `test_percentile_is_not_dimension_coordinate`?

Reply:
Hmm, no actually, it would make sense for the cube that we set up not to have that shape as it's never used.  Let me figure it out.